src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -856,25 +856,41 @@ public virtual bool HasUnsupportedMetadata
             }
         }
 
-        internal void ReportMissingNonNullTypesContextForAnnotation(DiagnosticBag diagnostics, Location location)
+        internal void ReportNullableReferenceTypesIfNeeded(DiagnosticBag diagnostics, Location location)
         {
-            ReportMissingNonNullTypesContextForAnnotation(this, diagnostics, location);
+            ReportNullableReferenceTypesIfNeeded(this.DeclaringCompilation, nonNullTypesContext: this, diagnostics, location);
         }
 
-        internal static void ReportMissingNonNullTypesContextForAnnotation(INonNullTypesContext context, DiagnosticBag diagnostics, Location location)
+        /// <summary>
+        /// A `?` annotation on a type that isn't a value type causes:
+        /// - an error before C# 8.0
+        /// - a warning outside of a NonNullTypes context
+        /// </summary>
+        internal static void ReportNullableReferenceTypesIfNeeded(CSharpCompilation compilation, INonNullTypesContext nonNullTypesContext, DiagnosticBag diagnostics, Location location)
         {
-            var diagnostic = ReportMissingNonNullTypesContextForAnnotation(context);
+            var diagnostic = ReportNullableReferenceTypesIfNeeded(compilation, nonNullTypesContext);
             if (diagnostic != null)
             {
                 diagnostics.Add(diagnostic, location);
             }
         }
 
-        internal static DiagnosticInfo ReportMissingNonNullTypesContextForAnnotation(INonNullTypesContext context)
+        internal static DiagnosticInfo ReportNullableReferenceTypesIfNeeded(CSharpCompilation compilation, INonNullTypesContext nonNullTypesContext)
```

Comment:
Giggling a bit at the method to determine if non-null is enabled for this context will eventually need a `?` annotation once we dogfood it in the compiler.